docs/src/migration-to-register-tool.md

```diff
@@ -0,0 +1,45 @@
+# Migration to `register_tool`
+
+This document applies to [PR#926](https://github.com/EmbarkStudios/rust-gpu/pull/926)
+
+## What happened
+
+In a [recent nightly Rust update](https://github.com/rust-lang/rust/commit/76dd5c58a011bb734ad5b8e96fc560374893bc8f), the `register_attr` feature was removed in favor of `register_tool`. Unfortunately, rust-gpu made use of this feature to register the `spirv` attribute.
+
+## What does this mean for you as a shader maintainer
+
+You'll need to import the `spirv` proc macro attribute from `spirv-std` in order for the rust compiler:
+
+```rust
+use spirv_std::spirv;
+```
+
+If your shader code already contains this line but is conditionally only included for non-spirv builds, like so:
+
+```rust
+#[cfg(not(target_arch = "spirv"))]
+use spirv_std::spirv;
+```
+
+please remove the conditional attribute.
+
+For this macro attribute to work correctly, it is important that `spirv` is visible in the global score and you use it like you used it before: `#[spirv(..)]`. An attempt to scope the attribute (such as `#[spirv_std::spirv(..)]`) will confuse the macro and likely fail.
```

Comment:
We don't need to expand on this much more but to be clear this only applies to weird "leaf" things like function args, fields, etc. (which are replaced *while* transforming a larger thing that triggered the proc macro as usual).
Whole-definition attributes (e.g. the whole entry-point or a whole `struct`) will just work through normal name resolution.

Funny thing about that btw, if you *only* need it for a leaf, you need to do:
```rust
#[spirv()]
fn foo(
    #[spirv(ray_tracing_weirdness)] x: T,
) {}
```
(with the whole-`fn` one only serving to handle the inner ones lol - again, I don't think we have to mention this, since we don't have that many attributes to begin with, and so far such a usecase doesn't exist AFAIK, but I wanted to expand on it "for the record" so to speak)

Reply:
If we do end up having a case where this applies (I don't think we do currently?), I'd suggest supporting `#[spirv]` without the parens.

Reply:
> I'd suggest supporting `#[spirv]` without the parens.

I don't think we have a say in the matter (i.e. both shouldwork), and you likely can't tell them apart (since that's the proc macro, not the tool module one).